frontend/packages/policy-editor/src/components/PolicyCardRules/PolicyRule/PolicyAccessPackages/PolicyAccessPackages.tsx

```diff
@@ -0,0 +1,180 @@
+  };
+
+  const handleAccessPackageChange = (newSelectedAccessPackageUrns: string[]): void => {
+    const updatedRules = getUpdatedRules(
+      {
+        ...policyRule,
+        accessPackages: newSelectedAccessPackageUrns,
+      },
+      policyRule.ruleId,
+      policyRules,
+    );
+    setPolicyRules(updatedRules);
+    savePolicy(updatedRules);
+  };
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setSearchValue(event.target.value);
+  };
+
+  const accessPackagesToRender = filterAccessPackagesBySearchString(
+    groupedAccessPackagesByArea,
+    searchValue,
+  );
+
+  return (
+    <div className={classes.accessPackages}>
+      <StudioAlert severity='warning' size='sm'>
+        <StudioLabelAsParagraph size='md' spacing>
+          {t('policy_editor.access_package_warning_header')}
+        </StudioLabelAsParagraph>
+        <StudioParagraph size='sm'>
+          {t('policy_editor.access_package_warning_body')}
+        </StudioParagraph>
+      </StudioAlert>
+      <StudioLabelAsParagraph size='md' spacing>
+        {t('policy_editor.access_package_header')}
+      </StudioLabelAsParagraph>
+      <ChosenAccessPackages
+        chosenAccessPackages={chosenAccessPackages}
+        groupedAccessPackagesByArea={groupedAccessPackagesByArea}
+        handleSelectAccessPackage={handleSelectAccessPackage}
+      />
+      <StudioLabelAsParagraph size='xs' spacing>
+        {t('policy_editor.access_package_all_packages')}
+      </StudioLabelAsParagraph>
+      <StudioTextfield
+        label={
+          <StudioLabelAsParagraph size='xs'>
+            {t('policy_editor.access_package_search')}
+          </StudioLabelAsParagraph>
+        }
+        hideLabel
+        placeholder={t('policy_editor.access_package_search')}
+        size='small'
+        value={searchValue}
+        onChange={handleSearch}
+      />
+      <AllAccessPackages
+        chosenAccessPackages={chosenAccessPackages}
+        accessPackagesToRender={accessPackagesToRender}
+        searchValue={searchValue}
+        handleSelectAccessPackage={handleSelectAccessPackage}
+      />
+    </div>
+  );
+};
+
+interface ChosenAccessPackagesProps {
+  chosenAccessPackages: string[];
+  groupedAccessPackagesByArea: PolicyAccessPackageArea[];
+  handleSelectAccessPackage: (accessPackageUrn: string) => void;
+}
+const ChosenAccessPackages = ({
+  chosenAccessPackages,
+  groupedAccessPackagesByArea,
+  handleSelectAccessPackage,
+}: ChosenAccessPackagesProps): ReactElement => {
+  const { t } = useTranslation();
+
+  if (chosenAccessPackages.length > 0) {
+    return (
+      <>
+        <StudioLabelAsParagraph size='xs' spacing>
+          {t('policy_editor.access_package_chosen_packages')}
+        </StudioLabelAsParagraph>
+        {groupedAccessPackagesByArea
+          .flatMap((area) => area.packages)
+          .filter((accessPackage) => chosenAccessPackages.includes(accessPackage.urn))
+          .map((accessPackage) => {
```

Comment:
Maybe we can refactor this to increase readability? 😄 
And we need some better names than what I suggested 😅 

```tsx
...
const flatMappedList: type = flatMapList(groupedAccessPackagesByArea);
const filteredList: type = filterOutSomething(flatMappedList);
...
   <PolicyAccessPackageAccordions list={filteredList} />
...

type PolicyAccessPackageAccordionsProps = {
   list: type;
   handleSelectAccessPackage: (accessPackageUrn: string) => void;
}
const PolicyAccessPackageAccordions = ({ list, handleSelectAccessPackage }: PolicyAccessPackageAccordionsProps): ReactElement[] => {
   return list.map((accessPackage: type) => (
      <PolicyAccessPackageAccordion
         key={accessPackage.urn}
         accessPackage={accessPackage}
         isChecked={true}
         handleSelectChange={handleSelectAccessPackage}
      />
   (
}

...

const flatMapList = (list: type): type => {
   return list.flatMap((area: type) => area.packages);
}

const filterOutSomething = (list: type, chosenAccessPackages: type): type => {
   return list.filter((accessPackage: type) => chosenAccessPackages.includes(accessPackage.urn))
}
```